python_modules/dagster/dagster/core/instance/__init__.py

```diff
@@ -1795,6 +1799,15 @@ def __exit__(self, exception_type, exception_value, traceback):
         if DagsterInstance._EXIT_STACK:
             DagsterInstance._EXIT_STACK.close()
 
+    def __del__(self):
+        if not self._disposed:
+            warnings.warn(
+                "Deleting a DagsterInstance object without disposing it first - DagsterInstance "
+                "should be created in a contextmanager to ensure that any resources that it "
+                "creates are always cleaned up"
+            )
```

Comment:
i suspect that this will add a great deal of noise. For example all our test apis default in a `DagsterInstance.ephemeral()` that as far as i know doesn't have any resources to clean. 

At least gate this to persistent type instances, maybe drop it.